projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -215,4 +215,22 @@ public function add_theme_install_menu( $wp_admin = false ) {
 
 		add_submenu_page( $parent_menu_slug, esc_attr__( 'Add New Theme', 'jetpack' ), __( 'Add New Theme', 'jetpack' ), 'install_themes', 'theme-install.php', null, 1 );
 	}
+
+	/**
+	 * Adds Users menu.
+	 *
+	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
+	 */
+	public function add_users_menu( $wp_admin = false ) {
+		parent::add_users_menu( $wp_admin );
```

Comment:
Just wondering, are there any performance or cognitive gains of using the inheritance here instead of just creating a new method that simply adds the menu item in question like here https://github.com/Automattic/jetpack/pull/18780/files#diff-49661f625194f9a0604aecd4f8723215bdf82e9d63d7cbfefbbf977a90bebe5cR41-R217?

Reply:
The problem with a separated method is that it actually depends on another method. In your example, calling `add_theme_install_menu` without calling `add_appearance_menu` first will result in the submenu not being added, forcing developers to remember that the two methods should be called sequentially as we needed to do in the unit test: https://github.com/Automattic/jetpack/blob/7c93abdede8a7b64694a7bd0534db6a3ed352a79/projects/plugins/jetpack/tests/php/modules/masterbar/test-class-atomic-admin-menu.php#L348-L349

For dependencies like that, I prefer to use a inheritance approach.

Reply:
Very good point, thanks for taking time to explain it!